### ISSUE TITLE
Styling points on mark type line

### ISSUE THREAD

**Issue by mcnuttandrew** [vega/vega-lite#6111]
The mark type line offers a boolean point property to indicate whether or not that vertices of the line should be decorated with circles. The circles created by the property seem to operate under a slightly different set of style property than expected. For instance this spec: 

```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v4.json",
  "description": "Google's stock price over time.",
  "data": {"url": "data/stocks.csv"},
  "transform": [{"filter": "datum.symbol==='GOOG'"}],
  "mark": {"type": "line", "point": true, "opacity": 0.5, "color": "red"},
  "encoding": {
    "x": {"field": "date", "type": "temporal"},
    "y": {"field": "price", "type": "quantitative"}
  }
}
```

Generates this image

<img width="275" alt="Screen Shot 2020-03-15 at 5 07 35 PM" src="https://user-images.githubusercontent.com/6854312/76711586-80087700-66df-11ea-9197-c5225f0cd6ba.png">

It seems like the points should be red and at 0.5 opacity as they are part of the line mark (at least that's how it reads now). While I wouldn't be surprised if there was a way to style these points and i just wasn't aware of that method (vega-lite is surprisingly big!), it seems like that type of binding of mark and style property is intuitive?

**Comment by kanitw** [comment]
https://vega.github.io/vega-lite/docs/line.html#properties 

The `point` property can have other mark properties for the points.

For example: 

```
{
  "$schema": "https://vega.github.io/schema/vega-lite/v4.json",
  "description": "Google's stock price over time.",
  "data": {"url": "data/stocks.csv"},
  "transform": [{"filter": "datum.symbol==='GOOG'"}],
  "mark": {"type": "line", "point": {"color": "green"}, "opacity": 0.5, "color": "red"},
  "encoding": {
    "x": {"field": "date", "type": "temporal"},
    "y": {"field": "price", "type": "quantitative"}
  }
}
```

![image](https://user-images.githubusercontent.com/111269/76714098-77657000-66e2-11ea-90cc-1ed99b0cffaf.png)

**Comment by domoritz** [comment]
I agree that it would make sense to use the same color by default l, though.

**Comment by kanitw** [comment]
I see
